src/contracts/pods/EigenPodManager.sol

```diff
@@ -142,113 +147,41 @@ contract EigenPodManager is
-                _podOwnerUndelegationLimboStatus[msg.sender].delegatedAddress,
-                limboStartBlock,
-                middlewareTimesIndex
-            ),
-            "EigenPodManager.exitUndelegationLimbo: shares in limbo are still slashable"
-        );
-
-        // enforce minimum delay lag
-        require(
-            limboStartBlock + strategyManager.withdrawalDelayBlocks() <= block.number,
-            "EigenPodManager.exitUndelegationLimbo: withdrawalDelayBlocks period has not yet passed"
-        );
-
-        // delete the pod owner's undelegation limbo details
-        delete _podOwnerUndelegationLimboStatus[msg.sender];
-
-        // emit event
-        emit UndelegationLimboExited(msg.sender);
-
-        // either withdraw the funds entirely from EigenLayer
-        if (withdrawFundsFromEigenLayer) {
-            // store the pod owner's current share amount in memory, for gas efficiency
-            uint256 currentPodOwnerShares = podOwnerShares[msg.sender];
-            // remove the shares from the podOwner and reduce the `withdrawableRestakedExecutionLayerGwei` in the pod
-            podOwnerShares[msg.sender] = 0;
-            _decrementWithdrawableRestakedExecutionLayerGwei(msg.sender, currentPodOwnerShares);
-            // withdraw through the ETH from the EigenPod
-            _withdrawRestakedBeaconChainETH(msg.sender, msg.sender, currentPodOwnerShares);
-            // or else return the "shares" to the delegation system
-        } else {
-            delegationManager.increaseDelegatedShares(msg.sender, beaconChainETHStrategy, podOwnerShares[msg.sender]);
-        }
+    function addShares(
+        address podOwner,
+        uint256 shares
+    ) external onlyDelegationManager returns (uint256) {
+        return _addShares(podOwner, shares);
     }
 
-    /**
-     * @notice forces the podOwner into the "undelegation limbo" mode, and returns the number of virtual 'beacon chain ETH shares'
-     * that the podOwner has, which were entered into undelegation limbo.
-     * @param podOwner is the staker to be forced into undelegation limbo
-     * @param delegatedTo is the operator the staker is currently delegated to
-     * @dev This function can only be called by the DelegationManager contract
-     */
-    function forceIntoUndelegationLimbo(
-        address podOwner,
-        address delegatedTo
-    )
-        external
-        onlyDelegationManager
-        onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH)
-        onlyNotFrozen(podOwner)
-        nonReentrant
-        returns (uint256 sharesRemovedFromDelegation)
-    {
-        // put the `podOwner` into undelegation limbo, and return the amount of shares which were entered into undelegation limbo
-        return _enterUndelegationLimbo(podOwner, delegatedTo);
+    /// @notice Used by the DelegationManager to complete a withdrawal, sending tokens to some destination address
+    // TODO the 2 calls here can probably be combined?
```

Comment:
nice, we should do that!

Reply:
yes, done https://github.com/Layr-Labs/eigenlayer-contracts/pull/222/commits/c587b10684a8f37e68ae1787e3007bb64560dd80